packages/@aws-cdk/aws-cloudwatch/lib/alarm.ts

```diff
@@ -162,6 +168,9 @@ export class Alarm extends AlarmBase {
         extendedStatistic: renderIfExtendedStatistic(props.statistic),
       });
     }
+    if (ANOMALY_DETECTION_COMPARISON_OPERATORS.includes(comparisonOperator) && !props.thresholdMetricId) {
```

Comment:
Wouldn't this break existing customers using ant of the comparison operators in  `ANOMALY_DETECTION_COMPARISON_OPERATORS`?

Reply:
Following issue #10540, my understanding is that customers could not apply any of these anomaly detection comparison operators without a threshold metric ID. This leads me to believe it would not break any existing customer functionality - on the contrary, it's fixing a problem - but if I've misunderstood the problem I'll be happy to revisit this.